internal/health_checks/api_check.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+import (
+    "fmt"
+    "cloud.google.com/go/logging"
+    "github.com/GoogleCloudPlatform/ops-agent/confgenerator/resourcedetector"
+    "github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+    "context"
+
+    monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+)
+
+type APICheck struct{
+    HealthCheck
+}
+
+func (c APICheck) RunCheck(uc *confgenerator.UnifiedConfig) error {
+    var project string
+    c.LogMessage("Get MetadataResource : ")
+    MetadataResource, err := resourcedetector.GetResource()
+    if err != nil {
+        return fmt.Errorf("can't get resource metadata: %w", err)
+    }
+    if gceMetadata, ok := MetadataResource.(resourcedetector.GCEResource); ok {
+        c.LogMessage(fmt.Sprintf("==> gceMetadata : %+v \n \n", gceMetadata))
+        project = gceMetadata.Project
+    } else {
+        // Not on GCE
+        project = "Not-on-GCE"
+    }
+    c.LogMessage(fmt.Sprintf("==> project : %s \n \n", project))
+
+    c.LogMessage("\n> APICheck \n \n")
+    ctx := context.Background()
+
+    // New Logging Client
+    c.LogMessage("==> New Logging Client \n")
+    logClient, err := logging.NewClient(ctx, project)
+    if err != nil {
+        c.Fail("Logging Client didn't create successfully.", "Check the Logging API is enabled.")
```

Comment:
I think it might be a good idea to start creating a set of error codes for failure cases. A (probably not very well thought out) example would be something like a struct with an error code and it's associated human readable message, like:

```
{
    code: logging-api-disabled,
    message: "Logging API disabled"
    action: "Check the Logging API is enabled"
    isFatal: true
}
```


Having them all defined and listed out in a separate file allows us to:
1. Easily find and modify errors as the Ops Agent evolves
2. Quickly identify all the tests that could generate a particular error
3. **Have a list of all the possible errors that the Health Checks will produce**. This will come in handy when we need to integrate the results of the health checks to other systems (such as a Health Oracle, or a Health Dashboard)

Reply:
Great! I really like this solution and it's very in tune with the next things to add to the current code. I will start working on this. Just a couple of questions and details regarding this solution :

1. How should we differentiate between `check failures` and `errors` ? Is the value `isFatal` would show this difference ? Should we add another value to your proposed struct ?
2. This actually solves the problem **How to do write integration test that work reliably ?** since we only a have a limited amount of error codes to check. This also will make the testing in multiple environments reliable and easier to understand the results.